gammapy/datasets/make.py

```diff
@@ -356,6 +364,213 @@ def make_test_bg_cube_model(detx_range=Angle([-10., 10.], 'degree'),
 
     return bg_cube_model
+
+
+def make_test_dataset(fits_path, overwrite=False,
+                      observatory_name='HESS', n_obs=10,
+                      datestart=None, dateend=None,
+                      random_state='random-seed'):
+    """
+    Make a test dataset and save it to disk.
+
+    Uses:
+
+    * `~gammapy.datasets.make_test_observation_table` to generate an
+      observation table
+
+    * `~gammapy.datasets.make_test_evenlist` to generate an event list
+      and effective area table for each observation
+
+    * `~gammapy.obs.DataStore` to handle the file naming scheme;
+      currently only the H.E.S.S. naming scheme is supported
+
+    See also :ref:`background_make_models_datasets_for_testing`.
+
+    Parameters
+    ----------
+    fits_path : str
+        Path to store the files.
+    overwrite : bool, optional
+        Flag to remove previous datasets in fits_path (if existing).
+    observatory_name : str, optional
+        Name of the observatory; a list of choices is given in
+        `~gammapy.obs.observatory_locations`.
+    n_obs : int
+        Number of observations for the obs table.
+    datestart : `~astropy.time.Time`, optional
+        Starting date for random generation of observation start time.
+    dateend : `~astropy.time.Time`, optional
+        Ending date for random generation of observation start time.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}, optional
+        Defines random number generator initialisation.
+        Passed to `~gammapy.utils.random.get_random_state`.
+    """
+    # create output folder
+    outdir = fits_path
+    if not os.path.isdir(outdir):
+        os.mkdir(outdir)
+    else:
+        if overwrite:
+            # delete and create again
+            shutil.rmtree(outdir) # recursively
+            os.mkdir(outdir)
+        else:
+            # do not overwrite, hence exit
+            s_error = "Cannot continue: directory \'{}\' exists.".format(outdir)
+            raise RuntimeError(s_error)
+
+
+    # generate observation table
+    observation_table = make_test_observation_table(observatory_name=observatory_name,
+                                                    n_obs=n_obs,
+                                                    datestart=datestart, dateend=dateend,
+                                                    use_abs_time=False,
+                                                    random_state=random_state)
+
+    # save observation list to disk
+    outfile = outdir + '/runinfo.fits'
+    observation_table.write(outfile)
+
+    # create data store for the organization of the files
+    # using H.E.S.S.-like dir/file naming scheme
+    if observatory_name == 'HESS':
+        scheme = 'hess'
+    else:
+        s_error = "Warning! Storage scheme for {}".format(observatory_name)
+        s_error += "not implemented. Only H.E.S.S. scheme is available."
+        raise RuntimeError(s_error)
```

Comment:
This should be a `ValueError` (always use that to handle cases of invalid inputs).